package.json

```diff
@@ -10,12 +10,22 @@
     "asciidoctor-pdf": "bin/asciidoctor-pdf",
     "asciidoctor-web-pdf": "bin/asciidoctor-web-pdf"
   },
+  "pkg": {
+    "assets": [
+      "css/**/*.css",
+      "node_modules/@fortawesome/fontawesome-svg-core/*.css"
+    ],
+    "scripts": [
+      "lib/document/**/*.js"
```

Comment:
(question): This is not infer by default from `files`?

Reply:
I shall check, but I'm pretty sure we need at least `lib/document/**/*.js`.
For other assets it depends how they're loaded and this is a bit of a catch all.

Reply:
If I don't include for instance Javascript files in `lib/document` I get:

```
23:01 $ ./asciidoctor-web-pdf 
ERROR:  Error: File '/**/asciidoctor-web-pdf/lib/document/repeating-table-elements.js' was not included into executable at compilation stage. Please recompile adding it as asset or script.
    at error_ENOENT (pkg/prelude/bootstrap.js:426:17)
    at readFileFromSnapshot (pkg/prelude/bootstrap.js:654:29)
    at Object.fs.readFileSync (pkg/prelude/bootstrap.js:698:18)
    at new DocumentPDFConverter (/snapshot/asciidoctor-web-pdf/lib/document/document-converter.js)
    at Object.<anonymous> (/snapshot/asciidoctor-web-pdf/lib/document/document-converter.js)
    at Module._compile (pkg/prelude/bootstrap.js:1320:22)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1156:10)
    at Module.load (internal/modules/cjs/loader.js:984:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1024:19) {
  errno: -2,
  code: 'ENOENT',
  path: '/snapshot/asciidoctor-web-pdf/lib/document/repeating-table-elements.js',
  pkg: true
}
```

Perhaps it could be inferrred if instead of doing something like:

```
this.repeatTableElementsContent = fs.readFileSync(`${__dirname}/repeating-table-elements.js`, 'utf8')
```

we instead did as the `pkg` docs I think require in the code:

```
path.join(__dirname, '../path/to/asset')
```

or perhaps `require.resolve` as the docs suggest:

>Hence, in order to make use of a file collected at packaging time (require a javascript file or serve an asset) you should take __filename, __dirname, process.pkg.defaultEntrypoint or require.main.filename as a base for your path calculations. For javascript files you can just require or require.resolve because they use current __dirname by default. For assets use path.join(__dirname, '../path/to/asset'). Learn more about path.join in Detecting assets in source code.

I'm not sure how much change should be made to other parts of the code to meet `pkg` requirements, perhaps it's better to have this in `package.json`?

Reply:
If I don't include css I get something like this:

```
23:23 $ ./asciidoctor-web-pdf ../../test.adoc
ERROR:  Error: File '/**/asciidoctor-web-pdf/css/asciidoctor.css' was not included into executable at compilation stage. Please recompile adding it as asset or script.
    at error_ENOENT (pkg/prelude/bootstrap.js:426:17)
    at readFileFromSnapshot (pkg/prelude/bootstrap.js:654:29)
    at Object.fs.readFileSync (pkg/prelude/bootstrap.js:698:18)
    at new DocumentPDFConverter (/snapshot/asciidoctor-web-pdf/lib/document/document-converter.js:37:39)
    at Object.<anonymous> (/snapshot/asciidoctor-web-pdf/lib/document/document-converter.js:503:18)
    at Module._compile (pkg/prelude/bootstrap.js:1320:22)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1156:10)
    at Module.load (internal/modules/cjs/loader.js:984:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1024:19) {
  errno: -2,
  code: 'ENOENT',
  path: '/snapshot/asciidoctor-web-pdf/css/asciidoctor.css',
  pkg: true
}
```

I don't think we want to try to deal with paths in dependencies...
It's just the font awesome one I'm a bit doubtful about so I've removed that.

I expect I introduced this on the basis of a simple `find`:

```
23:30 $ find . -name '*.css'
./css/document.css
./css/asciidoctor.css
./css/features/book.css
./css/features/title-document-numbering.css
./css/features/title-page.css
./examples/resume/style.css
./examples/letter/letter.css
./examples/cheat-sheet/snyk/assets/style.css
./examples/book/book.css
./examples/slides/ioslides/styles/main.css
./examples/slides/asciidoctor.css
./examples/slides/slides.css
./node_modules/@fortawesome/fontawesome-svg-core/styles.css
./node_modules/@asciidoctor/core/dist/css/asciidoctor.css
./node_modules/mocha/mocha.css
./node_modules/pngjs/coverage/lcov-report/prettify.css
./node_modules/pngjs/coverage/lcov-report/base.css
```

Reply:
FTR @Mogztter changed the path here.